nixos/modules/programs/sway.nix

```diff
@@ -121,16 +134,17 @@ in {
       }
     ];
     environment = {
-      systemPackages = [ swayPackage ] ++ cfg.extraPackages;
+      systemPackages = optional (cfg.package != null) cfg.package ++ cfg.extraPackages;
       # Needed for the default wallpaper:
-      pathsToLink = [ "/share/backgrounds/sway" ];
+      pathsToLink = optional (cfg.package != null) [ "/share/backgrounds/sway" ];
```

Comment:
```
nix-repl> :p lib.optional true [ "foo" ]
[ [ "foo" ] ]
```
A pretty common gotcha, should use `lib.optionals` (these namings are really confusing)

Reply:
Good catch, thanks :v: